packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -135,10 +135,23 @@ const ProductTile = (props) => {
                     }}
                 >
                     <IconButtonWithRegistration
-                        aria-label={intl.formatMessage({
-                            id: 'product_tile.assistive_msg.wishlist',
-                            defaultMessage: 'Wishlist'
-                        })}
+                        aria-label={
+                            isFavourite
+                                ? intl.formatMessage(
+                                      {
+                                          id: 'product_tile.assistive_msg.add_to_wishlist',
+                                          defaultMessage: 'Add {product} to wishlist'
+                                      },
+                                      {product: localizedProductName}
+                                  )
+                                : intl.formatMessage(
+                                      {
+                                          id: 'product_tile.assistive_msg.remove_from wishlist',
+                                          defaultMessage: 'Remove {product} from wishlist'
+                                      },
+                                      {product: localizedProductName}
+                                  )
+                        }
```

Comment:
The order of the messages should be reversed: if it's favourite, then clicking the icon means _removing_ the product from wishlist.